Sources/MessagingPush/CustomerIOParsedPushPayload.swift

```diff
@@ -29,19 +29,27 @@ public class CustomerIOParsedPushPayload {
 
     public var deepLink: URL? {
         get {
-            cio.push.link?.url
+            cio?.push.link?.url
         }
         set {
-            cioPush = cioPush.linkSet(newValue?.absoluteString)
+            if cioPush == nil {
+                cio = CioRichPushPayload(push: CioRichPushPayload.Push(link: newValue?.absoluteString, image: nil))
+            } else {
+                cioPush = cioPush?.linkSet(newValue?.absoluteString)
+            }
         }
     }
 
     public var image: URL? {
         get {
-            cio.push.image?.url
+            cio?.push.image?.url
         }
         set {
-            cioPush = cioPush.imageSet(newValue?.absoluteString)
+            if cioPush == nil {
+                cio = CioRichPushPayload(push: CioRichPushPayload.Push(link: nil, image: newValue?.absoluteString))
+            } else {
+                cioPush = cioPush?.imageSet(newValue?.absoluteString)
```

Comment:
can we add test for this? as codecov pointed out and i couldnt find this case in the test suite either

Reply:
Tests added. Thanks for the callout.